src/controllers/sendMessage.js

```diff
@@ -0,0 +1,78 @@
+import http from "http";
+
+import saveMessage from "../clients/saveMessage.js";
+import checkHasBudget from "../clients/checkHasBudget.js";
+import addToBudget from "../clients/addToBudget.js";
+
+export default async (req, res) => {
+  const body = JSON.stringify(req.body);
+
+  const postOptions = {
+    host: "127.0.0.1",
+    // host: "messageapp",
+    port: 3000,
+    path: "/message",
+    method: "post",
+    json: true,
+    headers: {
+      "Content-Type": "application/json",
+      "Content-Length": Buffer.byteLength(body),
+    },
+  };
+
+  const postReq = http.request(postOptions);
+
+  postReq.on("response", async (postRes) => {
+    try {
+      await addToBudget(-1);
+      await saveMessage({
+        ...req.body,
+        status: postRes.statusCode === 200 ? "OK" : "ERROR",
+      });
+      if (postRes.statusCode !== 200) {
+        throw new Error('Error in the messageapp request');
+      }
+
+      res.statusCode = 200;
+      res.end(postRes.body);
+    } catch (error) {
+      console.log(error.message);
+      res.statusCode = 500;
+      res.end(`Internal server error: SERVICE ERROR ${error.message}`);
+    }
+  });
+
+  postReq.on("timeout", async () => {
+    console.error("Timeout Exceeded!");
+    postReq.abort();
+
+    try {
+      await saveMessage({
+        ...req.body,
+        status: "TIMEOUT",
```

Comment:
It would be nice that on TIMEOUT or ERROR you handled the Credit state... if you charge them first you should do something with that if you get an error